packages/language-server/src/core/parseCSS.ts

```diff
@@ -1,169 +1,45 @@
 
-export function extractStylesheets(
-	snapshot: ts.IScriptSnapshot,
-	htmlDocument: HTMLDocument,
-	ast: ParseResult['ast']
-): VirtualCode[] {
-	const embeddedCSSCodes: VirtualCode[] = findEmbeddedStyles(snapshot, htmlDocument.roots);
-
-	const inlineStyles = findInlineStyles(ast);
-	if (inlineStyles.length > 0) {
-		const codes: Segment<CodeInformation>[] = [];
-		for (const inlineStyle of inlineStyles) {
-			codes.push('x { ');
-			codes.push([
-				inlineStyle.value,
-				undefined,
-				inlineStyle.position.start.offset + 'style="'.length,
-				{
-					completion: true,
-					verification: false,
-					semantic: true,
-					navigation: true,
-					structure: true,
-					format: false,
-				},
-			]);
-			codes.push(' }\n');
-		}
-
-		const mappings = buildMappings(codes);
-		const text = toString(codes);
-
-		embeddedCSSCodes.push({
-			id: 'inline.css',
-			languageId: 'css',
-			snapshot: {
-				getText: (start, end) => text.substring(start, end),
-				getLength: () => text.length,
-				getChangeRange: () => undefined,
-			},
-			embeddedCodes: [],
-			mappings,
-		});
-	}
-
-	return embeddedCSSCodes;
-}
-
-/**
- * Find all embedded styles in a document.
- * Embedded styles are styles that are defined in `<style>` tags.
- */
-function findEmbeddedStyles(snapshot: ts.IScriptSnapshot, roots: Node[]): VirtualCode[] {
-	const embeddedCSSCodes: VirtualCode[] = [];
-	let cssIndex = 0;
-
-	getEmbeddedCSSInNodes(roots);
-
-	function getEmbeddedCSSInNodes(nodes: Node[]) {
-		for (const [_, node] of nodes.entries()) {
-			if (
-				node.tag === 'style' &&
-				node.startTagEnd !== undefined &&
-				node.endTagStart !== undefined
-			) {
-				const styleText = snapshot.getText(node.startTagEnd, node.endTagStart);
-				embeddedCSSCodes.push({
-					id: `${cssIndex}.css`,
-					languageId: 'css',
-					snapshot: {
-						getText: (start, end) => styleText.substring(start, end),
-						getLength: () => styleText.length,
-						getChangeRange: () => undefined,
-					},
-					mappings: [
-						{
-							sourceOffsets: [node.startTagEnd],
-							generatedOffsets: [0],
-							lengths: [styleText.length],
-							data: {
-								verification: false,
-								completion: true,
-								semantic: true,
-								navigation: true,
-								structure: true,
-								format: false,
-							},
-						},
-					],
-					embeddedCodes: [],
-				});
-				cssIndex++;
-			}
-
-			if (node.children) getEmbeddedCSSInNodes(node.children);
-		}
-	}
-
-	return embeddedCSSCodes;
+export function extractStylesheets(styles: TSXExtractedStyle[]): VirtualCode {
+	return mergeCSSContexts(styles);
 }
 
-/**
- * Find all inline styles using the Astro AST
- * Inline styles are styles that are defined in the `style` attribute of an element.
- * TODO: Merge this with `findEmbeddedCSS`? Unlike scripts, there's no scoping being done here, so merging all of it in
- * the same virtual file is possible, though it might make mapping more tricky.
```

Comment:
This is the case now in the new format, which improves intellisense for class names and stuff when you have multiple style tags in the same file. Kinda unlikely, but still funny!